jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,249 @@
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/* Common function for optional argument */
+
+static jerry_value_t
+jerryx_arg_transform_optional(jerryx_arg_iterator_t *iterator_p,
+                              jerryx_arg_transform_func_t func)
+{
+  jerry_value_t peeked_js_arg = jerryx_arg_iterator_peek (iterator_p);
+
+  if (jerry_value_is_undefined (peeked_js_arg))
+  {
+    jerryx_arg_iterator_pop (iterator_p);
+    return jerry_create_undefined ();
+  }
+
+  return func(iterator_p);
+}
+
+
+/* Number transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_number_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_number (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a number.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_number_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_number_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_number = jerry_value_to_number (js_arg);
+
+  if (jerry_value_has_error_flag (to_number))
+  {
+    jerry_release_value (to_number);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a number.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_number_value (to_number);
+  jerry_release_value (to_number);
+
+  return jerry_create_undefined ();
+}
+
+/* Bool transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_boolean_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_boolean (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a boolean.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_boolean_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_boolean_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  bool to_boolean = jerry_value_to_boolean (js_arg);
+
+  bool *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = to_boolean;
+
+  return jerry_create_undefined ();
+}
+
+/* String transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_string_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_string (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a string.");
+  }
+
+
```

Comment:
The part below this shared with `jerryx_arg_transform_string_impl`, I would put this into a static helper to avoid the duplication.